src/onnx/mod.rs

```diff
@@ -102,9 +102,12 @@ impl Environment {
             .map_err(|_| NewSessionError::MalformedModelPath)?;
         unsafe {
             let allocator = self.api.get_allocator_with_default_options()?;
+            let sess_options = self.api.create_session_options()?;
+
+            sys::OrtSessionOptionsAppendExecutionProvider_CUDA(sess_options, 0);
```

Comment:
This fails to link on macOS. I imagine it would also fail to link on Linux if the runtime wasn't compiled with CUDA support.

Since we'll get the link error regardless of if we even invoke this function, I think the way to go here is to add a "cuda" feature.